### PR TITLE
Add semVerLevel to client HTTP requests (this is NuGet SemVer 2.0.0 protocol)

### PR DESCRIPTION
**Pull request by joelverhagen**
This implements [SemVer 2.0.0 Protocol](https://github.com/NuGet/Home/wiki/Semver-2.0.0-Protocol) in the client. In addition to the reviewed endpoints, I've added the autocomplete endpoints, which were overlooked in the initial design.

Registration blob SemVer 2.0.0 protocol support is already implemented by PR https://github.com/NuGet/NuGet.Client/pull/1094.

The basis of the design is that clients must opt-in to see SemVer 2.0.0 packages.

/cc @emgarten @alpaix @dtivel @skofman1 @rrelyea

### CODE REVIEW

File: src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs [hunk 1]
```diff
@@ -45,7 +44,8 @@ public AutoCompleteResourceV2Feed(HttpSourceResource httpSourceResource, string
         {
             var apiEndpointUri = new UriBuilder(new Uri(_baseUri, @"package-ids"))
             {
-                Query = $"partialId={packageIdPrefix}&includePrerelease={includePrerelease.ToString()}"
+                Query = $"partialId={packageIdPrefix}&includePrerelease={includePrerelease.ToString()}" +
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs]: nit: just add this to the string instead of using ``+``

**Review by emgarten — APPROVED** [review]
Looks great to me, glad this will be possible for V2 also! ✨

File: src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs (hunk 1 above)
Comment (alpaix) [reply src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs]: Is `ToString` needed here? Interpolated string implicitly converts all types.

Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs]: Nope, not needed. String interpolation does this by default, as you mentioned. It's pre-existing code but you're right, I should fix it up.

**Review by alpaix — APPROVED** [review]

**Comment by joelverhagen** [comment]
@rrelyea mentioned offline that this will not be in the 4.0.0 release. I will merge this only after `dev` or some other branches opens for the next release (4.0.1? Not determined yet).

File: src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGet.Core/NuGet.Protocol.Core.v3/LegacyFeed/AutoCompleteResourceV2Feed.cs]: Fixed.

**Review by alpaix — APPROVED** [review]

File: test/NuGet.Core.Tests/NuGet.Protocol.VisualStudio.Tests/PowershellAutoCompleteResourceTests.cs [hunk 2]
```diff
@@ -16,15 +21,54 @@ public class PowershellAutoCompleteResourceTests
         public PowershellAutoCompleteResourceTests()
         {
             ResponsesDict = new Dictionary<string, string>();
-            ResponsesDict.Add("http://testsource.com/v3/index.json", JsonData.IndexJson);
-            ResponsesDict.Add("https://api-v3search-0.nuget.org/autocomplete?q=elm", JsonData.PsAutoCompleteV3Example);
-            ResponsesDict.Add("https://nuget.org/api/v2/package-ids?partialId=elm", JsonData.PSAutoCompleteV2Example);
+            ResponsesDict.Add(
+                "http://testsource.com/v3/index.json",
```
Comment (dtivel) [inline test/NuGet.Core.Tests/NuGet.Protocol.VisualStudio.Tests/PowershellAutoCompleteResourceTests.cs]: How about http://source.test/v3/index.json instead?  [.test](https://en.wikipedia.org/wiki/.test) is a top-level domain name reserved for non-production use.  http://testsource.com is a real domain name that redirects to http://www.healthcaresource.com.  Although we may not make a network call here, using the .test top-level domain in tests whenever we need bogus URL's seems like a good practice.

**Review by dtivel — APPROVED** [review]

File: test/NuGet.Core.Tests/NuGet.Protocol.VisualStudio.Tests/PowershellAutoCompleteResourceTests.cs (hunk 2 above)
Comment (joelverhagen) [reply test/NuGet.Core.Tests/NuGet.Protocol.VisualStudio.Tests/PowershellAutoCompleteResourceTests.cs]: Good point. No point in accidentally hitting their web servers in the future!

Comment (joelverhagen) [reply test/NuGet.Core.Tests/NuGet.Protocol.VisualStudio.Tests/PowershellAutoCompleteResourceTests.cs]: Fixed.

**Comment by joelverhagen** [comment]
This PR address https://github.com/NuGet/Home/issues/3610 on the client side.

**Comment by emgarten** [comment]
This should go into dev (4.3.0) and possibly release-4.1.0-rtm for 4.1.0

**Comment by joelverhagen** [comment]
Rebased on `dev`, CI is green. @emgarten / @alpaix / @rrelyea - can I check in?

**Comment by alpaix** [comment]
LGTM

**Comment by joelverhagen** [comment]
FYI @xavierdecoster

**Pull request merged**